source/common/router/config_impl.cc

```diff
@@ -370,6 +370,10 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
   if (route.route().has_cors()) {
     cors_policy_ = std::make_unique<CorsPolicyImpl>(route.route().cors());
   }
+  for (const auto upgrade_config : route.route().upgrade_configs()) {
```

Comment:
In the HCM we worry about duplicate upgrade types, but it doesn't look like we do here. So afaict, last upgrade type boolean wins.